packages/shared/src/components/modals/common/ModalBody.tsx

```diff
@@ -0,0 +1,26 @@
+import classNames from 'classnames';
+import React, { ReactElement, ReactNode } from 'react';
+import { ModalKind, ModalPropsContext, ModalSize } from './types';
+
+export type ModalBodyProps = {
+  children?: ReactNode;
+};
+
+export function ModalBody({ children }: ModalBodyProps): ReactElement {
+  return (
+    <ModalPropsContext.Consumer>
+      {({ kind, size }) => (
+        <section
+          className={classNames(
+            'overflow-auto relative w-full h-full shrink max-h-full',
+            kind === ModalKind.FlexibleTop && size === ModalSize.Large
```

Comment:
This might be more readable to abstract outside the render as a condition?

Reply:
Done